cmake/modules/FindPython.cmake

```diff
@@ -29,11 +29,12 @@ else()
   list(APPEND PYTHON_LIBRARIES ${PC_PYTHON_STATIC_LIBRARIES})
 endif()
 
+list(APPEND PYTHON_DEFINITIONS -DHAS_PYTHON=1)
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(PYTHON REQUIRED_VARS PYTHON_INCLUDE_DIR PYTHON_LIBRARY PYTHON_LIBRARIES)
+find_package_handle_standard_args(PYTHON REQUIRED_VARS PYTHON_INCLUDE_DIR PYTHON_LIBRARY PYTHON_LIBRARIES PYTHON_DEFINITIONS)
 if(PYTHON_FOUND)
   set(PYTHON_INCLUDE_DIRS ${PYTHON_INCLUDE_DIR})
 endif()
 
-mark_as_advanced(PYTHON_EXECUTABLE PYTHON_INCLUDE_DIRS PYTHON_INCLUDE_DIR PYTHON_LIBRARY PYTHON_LIBRARIES PYTHON_LDFLAGS FFI_LIBRARY EXPAT_LIBRARY INTL_LIBRARY GMP_LIBRARY)
+mark_as_advanced(PYTHON_EXECUTABLE PYTHON_INCLUDE_DIRS PYTHON_INCLUDE_DIR PYTHON_LIBRARY PYTHON_LIBRARIES PYTHON_LDFLAGS FFI_LIBRARY EXPAT_LIBRARY INTL_LIBRARY GMP_LIBRARY PYTHON_DEFINITIONS)
```

Comment:
PYTHON_DEFINITIONS is not a cached variable (and shouldn't be one), so it doesn't make sense to mark it as advanced.